crs_application/src/crs_executive.cpp

```diff
@@ -0,0 +1,457 @@
+    static const std::string EXEC_MEDIA_CHANGE = "Exec_Media_Change";
+    static const std::string CHECK_QUEUE = "PE_Check_Queue";
+    static const std::string EXEC_HOME = "Exec_Home";
+  }
+}
+
+namespace crs_application
+{
+
+CRSExecutive::CRSExecutive(std::shared_ptr<rclcpp::Node> node):
+    node_(node)
+{
+  if(!setup())
+  {
+    throw std::runtime_error("Failed initial setup");
+  }
+}
+
+CRSExecutive::~CRSExecutive()
+{
+
+}
+
+common::ActionResult CRSExecutive::initialize()
+{
+  if(scan_acqt_mngr_->init() &&
+      motion_planning_mngr_->init() &&
+      part_regt_mngr_->init() &&
+      process_exec_mngr_->init() &&
+      part_rework_mngr_->init() )
+  {
+    RCLCPP_INFO(node_->get_logger(),"%s All managers initialized");
+  }
+  else
+  {
+    RCLCPP_ERROR(node_->get_logger(),"%s One or more managers failed to initialize");
+    return false;
+  }
+
+  RCLCPP_WARN(node_->get_logger(),"%s not fully implemented yet",__PRETTY_FUNCTION__);
+  return true;
+}
+
+common::ActionResult CRSExecutive::configure()
+{
+  using namespace crs_msgs::srv;
+  GetConfiguration::Request::SharedPtr req = std::make_shared<GetConfiguration::Request>();
+  std::shared_future < GetConfiguration::Response::SharedPtr > res = get_config_client_->async_send_request(req);
+
+  if (rclcpp::spin_until_future_complete(node_->get_node_base_interface(), res,
+                                         std::chrono::duration<double>(WAIT_SERVICE_COMPLETION_PERIOD))
+      != rclcpp::executor::FutureReturnCode::SUCCESS)
+  {
+    RCLCPP_ERROR(node_->get_logger(), "%s Failed to get configuration",node_->get_name());
+    return false;
+  }
+  if(!res.get()->success)
+  {
+    RCLCPP_ERROR(node_->get_logger(), "%s configution failed with err msg: ",node_->get_name(), res.get()->err_msg.c_str());
+    return false;
+  }
+
+  process_config_ = res.get()->config;
+  RCLCPP_ERROR(node_->get_logger(), "%s Got Configuration",node_->get_name());
+  return configureManagers();
+}
+
+
+bool CRSExecutive::configureManagers()
+{
+  // TODO: populate each config structure from the data in process_config_
+
+  return scan_acqt_mngr_->configure(task_managers::ScanAcquisitionConfig{}) &&
+      motion_planning_mngr_->configure(task_managers::MotionPlanningConfig{}) &&
+      part_regt_mngr_->configure(task_managers::PartRegistrationConfig{}) &&
+      process_exec_mngr_->configure(task_managers::ProcessExecutionConfig{}) &&
+      part_rework_mngr_->configure(task_managers::PartReworkConfig{});
+}
+
+bool CRSExecutive::addStateCallbacks(const std::map<std::string, StateCallbackInfo >& st_callbacks_map)
+{
+  using namespace crs_application;
+  using namespace scxml_core;
+
+  auto sm_ = this->sm_;
+  for(const auto& kv : st_callbacks_map)
+  {
+    // create actual callback
+    auto sm_entry_cb = [&sm_, &kv](const Action& action) -> Response
+    {
+      auto res = kv.second.entry_cb();
+      if(!res)
+      {
+        sm_->postAction(Action{.id= kv.second.on_failed_action});
```

Comment:
Same warning here. Action needs a constructor

Reply:
This is supposed to be a legal initialization, I don't see why it requires a constructor.

Reply:
The warning has been silenced now with the -Wno-missing-field-initializers compile option